spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/client/OAuth2ClientAutoConfiguration.java

```diff
@@ -0,0 +1,182 @@
+
+	static Set<String> getClientKeys(Environment environment) {
+		return getClientPropertiesByClient(environment).keySet();
+	}
+
+	static Map<String, Map> getClientPropertiesByClient(Environment environment) {
+		Map<String, Object> clientPropertiesByClient = Binder.get(environment)
+				.bind(CLIENT_PROPERTY_PREFIX, Bindable.mapOf(String.class, Object.class))
+				.orElse(new HashMap<>());
+
+		// Filter out clients that don't have the client-id property set
+		return clientPropertiesByClient.entrySet().stream()
+				.map(e -> new AbstractMap.SimpleImmutableEntry<>(e.getKey(), (Map) e.getValue()))
+				.filter(e -> e.getValue().containsKey(CLIENT_ID_PROPERTY))
+				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+	}
+
+	@Order(1)
+	@Configuration
+	@Conditional(ClientsConfiguredCondition.class)
+	@ConditionalOnMissingBean(ClientRegistrationRepository.class)
+	protected static class ClientRegistrationRepositoryConfiguration {
+		private final Environment environment;
+
+		protected ClientRegistrationRepositoryConfiguration(Environment environment) {
+			this.environment = environment;
+		}
+
+		@Bean
+		public ClientRegistrationRepository clientRegistrationRepository() {
+			List<ClientRegistration> clientRegistrations = new ArrayList<>();
+
+			Binder binder = Binder.get(this.environment);
+			getClientKeys(this.environment).forEach(clientKey -> {
+				String fullClientKey = CLIENT_PROPERTY_PREFIX + "." + clientKey;
+				ClientRegistrationProperties clientRegistrationProperties = binder.bind(
+						fullClientKey, Bindable.of(ClientRegistrationProperties.class)).get();
+				clientRegistrations.add(new ClientRegistration.Builder(clientRegistrationProperties).build());
+			});
+
+			return new InMemoryClientRegistrationRepository(clientRegistrations);
+		}
+	}
+
+	@Order(2)
+	@ConditionalOnMissingBean(WebSecurityConfiguration.class)
+	@ConditionalOnBean(ClientRegistrationRepository.class)
+	@EnableWebSecurity
+	protected static class OAuth2LoginConfiguration extends WebSecurityConfigurerAdapter {
+		private final Environment environment;
+		private final ClientRegistrationRepository clientRegistrationRepository;
+
+		protected OAuth2LoginConfiguration(Environment environment, ClientRegistrationRepository clientRegistrationRepository) {
+			this.environment = environment;
+			this.clientRegistrationRepository = clientRegistrationRepository;
+		}
+
+		// @formatter:off
+		@Override
+		protected void configure(HttpSecurity http) throws Exception {
+			http
+				.authorizeRequests()
+					.anyRequest().authenticated()
+					.and()
+				.oauth2Login()
+					.clients(this.clientRegistrationRepository);
+
+			this.registerUserNameAttributeNames(http.oauth2Login());
+		}
+		// @formatter:on
+
+		private void registerUserNameAttributeNames(OAuth2LoginConfigurer<HttpSecurity> oauth2LoginConfigurer) throws Exception {
+			getClientPropertiesByClient(this.environment).entrySet().stream().forEach(e -> {
```

Comment:
`.forEach()` rather than `.stream().forEach()` would work here wouldn't it?

Reply:
Yes it would. Will update.